node/tests/SharedTests.ts

```diff
@@ -161,7 +161,19 @@ export function runBaseTests(config: {
         async (protocol) => {
             await runTest(
                 async (client: BaseClient) => {
-                    expect(await client.clientGetName()).toBe("TEST_CLIENT");
+                    expect(await client.clientGetName()).toEqual("TEST_CLIENT");
+
+                    if (client instanceof GlideClient) {
```

Comment:
does it look better if we use a conditional? 
```
if (client instanceof GlideClient) {
expect(await client.clientGetName(
    client instanceof GlideClient ?
        Decoder.Bytes :
        { decoder: Decoder.Bytes }
    ).toEqual(Buffer.from("TEST_CLIENT"));
```

Reply:
Unfortunately
```
Argument of type 'Decoder.Bytes | { decoder: Decoder.Bytes; }' is not assignable to parameter of type '(Decoder & { route?: Routes | undefined; decoder?: Decoder | undefined; }) | undefined'.
  Type '{ decoder: Decoder.Bytes; }' is not assignable to type '(Decoder & { route?: Routes | undefined; decoder?: Decoder | undefined; }) | undefined'.
    Type '{ decoder: Decoder.Bytes; }' is not assignable to type 'Decoder.String & { route?: Routes | undefined; decoder?: Decoder | undefined; }'.
      Type '{ decoder: Decoder.Bytes; }' is not assignable to type 'Decoder.String'.ts(2345)
```